src/sentry/api/endpoints/api_application_rotate_secret.py

```diff
@@ -0,0 +1,27 @@
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.models import ApiApplication, ApiApplicationStatus
+from sentry.models.apiapplication import generate_token
+
+
+@control_silo_endpoint
+class ApiApplicationRotateSecretEndpoint(Endpoint):
```

Comment:
You're going to want to split out your backend from frontend change.

Once you take this out of draft you're going to get a comment from a bot telling you this

	  🚨 **Warning:** This pull request contains Frontend and Backend changes!
	
	  It's discouraged to make changes to Sentry's Frontend and Backend
	  in a single pull request. The Frontend and Backend are **not**
	  atomically deployed. If the changes are interdependent of each
	  other, they **must** be separated into two pull requests and be made
	  forward or backwards compatible, such that the Backend or Frontend
	  can be safely deployed independently.
	
	
	  Have questions? Please ask in the [`#discuss-dev-infra`
	  channel](https://app.slack.com/client/T024ZCV9U/CTJL7358X).

You'll want to merge you're backend change first and deploy it.

Reply:
backend merge deploying now - gonna rebase after so it'll just be FE

Reply:
👍